code/controllers/subsystem/mobs.dm

```diff
@@ -1,14 +1,21 @@
 SUBSYSTEM_DEF(mobs)
 	name = "Mobs"
 	priority = 100
-	flags = SS_KEEP_TIMING|SS_NO_INIT
+	flags = SS_KEEP_TIMING
 	runlevels = RUNLEVEL_GAME | RUNLEVEL_POSTGAME
 
 	var/list/currentrun = list()
+	var/static/list/by_zlevel = list()
 
 /datum/controller/subsystem/mobs/stat_entry()
 	..("P:[GLOB.mob_living_list.len]")
 
+/datum/controller/subsystem/mobs/Initialize(start_timeofday)
+	by_zlevel.len = world.maxz
+	for (var/i = 1 to world.maxz)
+		by_zlevel.[i] = list()
```

Comment:
and this compiles because byond is stupid

Reply:
The worst part is that this was the only way I managed to build a variable-length multidimensional list. 

None of the documented ways to initializing a multidimensional list work with variables or work outside of _init() (where maxz is 1 due to maps not being loaded yet), and the usual suspects don't work because insertion functions merge lists so attempting to insert an empty list just does nothing.

At least .len is intentionally read/write and modifying it is documented and supported and it doesn't blow up.

Reply:
No no, what you've done here is fine.
I'm talking about ```by_zlevel.[i]```
that's ```.[]``` which isn't an operator, it's two, var lookup and indexing, why it's compiling I haven't a clue besides byond~

Reply:
Oh fuck, that's actually a typo/left-over `.` from .Insert/.Add tests.

I have no idea why it compiles or works either. I'll remove it when i get home.